### PR TITLE
Store behaviour docs in the byte code

### LINKED ISSUE #3088: Behaviour docs should be stored in the byte code too

### PR DESCRIPTION
**Pull request by lexmag**
Closes #3088

### CODE REVIEW

File: lib/elixir/lib/behaviour.ex [hunk 1]
```diff
@@ -127,26 +127,17 @@ defmodule Behaviour do
...
-      :lists.reverse Module.get_attribute(env.module, :behaviour_docs)
-    end
-
-    quote do
       @doc false
       def __behaviour__(:callbacks) do
         __MODULE__.behaviour_info(:callbacks)
       end
 
       def __behaviour__(:docs) do
```
Comment (lexmag) [inline lib/elixir/lib/behaviour.ex]: Shall we deprecate/remove this function clause?

File: lib/elixir/lib/code.ex [hunk 2]
```diff
@@ -532,8 +532,10 @@ defmodule Code do
   # unsupported chunk version
   defp lookup_docs(_, _), do: nil
 
+  @doc_sections [:docs, :moduledoc, :behaviour_docs]
```
Comment (josevalim) [inline lib/elixir/lib/code.ex]: I think we need to update the docs for `Code.get_docs/1`.

File: lib/elixir/lib/behaviour.ex (hunk 1 above)
Comment (josevalim) [inline lib/elixir/lib/behaviour.ex]: Just add two TODOs like these please: https://github.com/elixir-lang/elixir/blob/master/lib/elixir/lib/regex.ex#L634-L635

**Comment by lexmag** [comment]
Updated.

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Comment by josevalim** [comment]
This is awesome, thank you!

**Comment by lexmag** [comment]
Thank you!
I think we need make an IEx helper for it from now on, right?

**Comment by josevalim** [comment]
Sure! Suggestions on how the helper should be called? `b` for Behaviour?

**Comment by lexmag** [comment]
Guess so, since `c`(for callback) isn't available `b` nicely fits to the `s`(pec) and `t`(ype) group.

**Pull request merged**